python/pyarrow/_dataset_parquet.pyx

```diff
@@ -808,6 +815,18 @@ cdef class ParquetFragmentScanOptions(FragmentScanOptions):
             raise ValueError("size must be larger than zero")
         self.reader_properties().set_thrift_container_size_limit(size)
 
+    def set_file_decryption_properties(self, FileDecryptionProperties decryption_properties):
```

Comment:
Several things:
1) will this work if Parquet encryption is not enabled at compile time? (see how `parquet_decryption_config` works)
2) can you make this a read/write property like the other options here?
3) can it also be added to the constructor like the other options?
4) can you add unit tests for it? see existing test in https://github.com/apache/arrow/blob/40a8a68c7c3903e2bad98605d82339460c5ea930/python/pyarrow/tests/test_dataset_encryption.py#L131
5) how does this interact with `decryption_config`?

Reply:
@pitrou is this approximately what you had in mind?
```diff
diff --git a/python/pyarrow/_dataset_parquet.pxd b/python/pyarrow/_dataset_parquet.pxd
index d5bc172d3..980f645bc 100644
--- a/python/pyarrow/_dataset_parquet.pxd
+++ b/python/pyarrow/_dataset_parquet.pxd
@@ -29,6 +29,7 @@ cdef class ParquetFragmentScanOptions(FragmentScanOptions):
     cdef:
         CParquetFragmentScanOptions* parquet_options
         object _parquet_decryption_config
+        object _file_decryption_properties
 
     cdef void init(self, const shared_ptr[CFragmentScanOptions]& sp)
     cdef CReaderProperties* reader_properties(self)
diff --git a/python/pyarrow/_dataset_parquet.pyx b/python/pyarrow/_dataset_parquet.pyx
index 9d97a7e34..3f774fb83 100644
--- a/python/pyarrow/_dataset_parquet.pyx
+++ b/python/pyarrow/_dataset_parquet.pyx
@@ -56,7 +56,7 @@ from pyarrow._parquet cimport (
 
 try:
     from pyarrow._dataset_parquet_encryption import (
-        set_encryption_config, set_decryption_config
+        set_encryption_config, set_decryption_config, set_file_decryption_properties
     )
     parquet_encryption_enabled = True
 except ImportError:
@@ -132,8 +132,7 @@ cdef class ParquetFileFormat(FileFormat):
             default_fragment_scan_options = ParquetFragmentScanOptions(**scan_args)
             # make sure scan options has decryption properties
             if decryption_properties is not None:
-                default_fragment_scan_options.set_file_decryption_properties(
-                    decryption_properties)
+                default_fragment_scan_options.file_decryption_properties = decryption_properties
 
         elif isinstance(default_fragment_scan_options, dict):
             default_fragment_scan_options = ParquetFragmentScanOptions(
@@ -721,6 +720,9 @@ cdef class ParquetFragmentScanOptions(FragmentScanOptions):
     decryption_config : pyarrow.dataset.ParquetDecryptionConfig, default None
         If not None, use the provided ParquetDecryptionConfig to decrypt the
         Parquet file.
+    file_decryption_properties : pyarrow.dataset.FileDecryptionProperties, default None
+        If not None, use the provided FileDecryptionProperties to decrypt encrypted
+        Parquet file.
     page_checksum_verification : bool, default False
         If True, verify the page checksum for each page read from the file.
     """
@@ -735,6 +737,7 @@ cdef class ParquetFragmentScanOptions(FragmentScanOptions):
                  thrift_string_size_limit=None,
                  thrift_container_size_limit=None,
                  decryption_config=None,
+                 file_decryption_properties=None,
                  bint page_checksum_verification=False):
         self.init(shared_ptr[CFragmentScanOptions](
             new CParquetFragmentScanOptions()))
@@ -749,6 +752,8 @@ cdef class ParquetFragmentScanOptions(FragmentScanOptions):
             self.thrift_container_size_limit = thrift_container_size_limit
         if decryption_config is not None:
             self.parquet_decryption_config = decryption_config
+        if file_decryption_properties is not None:
+            self.file_decryption_properties = file_decryption_properties
         self.page_checksum_verification = page_checksum_verification
 
     cdef void init(self, const shared_ptr[CFragmentScanOptions]& sp):
@@ -818,16 +823,24 @@ cdef class ParquetFragmentScanOptions(FragmentScanOptions):
             raise ValueError("size must be larger than zero")
         self.reader_properties().set_thrift_container_size_limit(size)
 
-    def set_file_decryption_properties(self, FileDecryptionProperties decryption_properties):
-        """
-        Set the file decryption properties for the Parquet fragment scan options.
-        Parameters
-        ----------
-        decryption_properties : FileDecryptionProperties
-            The decryption properties to be used for reading encrypted Parquet files.
-        """
-        cdef CReaderProperties* reader_props = self.reader_properties()
-        reader_props.file_decryption_properties(decryption_properties.unwrap())
+    @property
+    def file_decryption_properties(self):
+        if not parquet_encryption_enabled:
+            raise NotImplementedError(
+                "Unable to access encryption features. "
+                "Encryption is not enabled in your installation of pyarrow."
+            )
+        return self._file_decryption_properties
+
+    @file_decryption_properties.setter
+    def file_decryption_properties(self, config):
+        if not parquet_encryption_enabled:
+            raise NotImplementedError(
+                "Encryption is not enabled in your installation of pyarrow, but "
+                "file_decryption_properties were provided."
+            )
+        set_file_decryption_properties(self, config)
+        self._file_decryption_properties = config
 
     @property
     def parquet_decryption_config(self):
diff --git a/python/pyarrow/_dataset_parquet_encryption.pyx b/python/pyarrow/_dataset_parquet_encryption.pyx
index 11a7174eb..0e1dbfeb4 100644
--- a/python/pyarrow/_dataset_parquet_encryption.pyx
+++ b/python/pyarrow/_dataset_parquet_encryption.pyx
@@ -162,6 +162,14 @@ def set_encryption_config(
     opts.parquet_options.parquet_encryption_config = c_config
 
 
+def set_file_decryption_properties(
+    ParquetFragmentScanOptions opts not None,
+    FileDecryptionProperties config not None
+):
+    cdef CReaderProperties* reader_props = opts.reader_properties()
+    reader_props.file_decryption_properties(config.unwrap())
+
+
 def set_decryption_config(
     ParquetFragmentScanOptions opts not None,
     ParquetDecryptionConfig config not None
```

Reply:
Perhaps tests could be refactored like so:
```diff
diff --git a/python/pyarrow/tests/parquet/test_encryption.py b/python/pyarrow/tests/parquet/test_encryption.py
index 7fc414381..3ccb4cd02 100644
--- a/python/pyarrow/tests/parquet/test_encryption.py
+++ b/python/pyarrow/tests/parquet/test_encryption.py
@@ -100,6 +100,8 @@ def write_encrypted_file(path, data_table, footer_key_name, col_key_name,
     write_encrypted_parquet(path, data_table, encryption_config,
                             kms_connection_config, crypto_factory)
 
+    return kms_connection_config, crypto_factory
+
 
 def test_encrypted_parquet_write_read(tempdir, data_table):
     """Write an encrypted parquet, verify it's encrypted, and then read it."""
@@ -117,17 +119,12 @@ def test_encrypted_parquet_write_read(tempdir, data_table):
         cache_lifetime=timedelta(minutes=5.0),
         data_key_length_bits=256)
 
-    write_encrypted_file(path, data_table, FOOTER_KEY_NAME, COL_KEY_NAME,
-                         FOOTER_KEY, COL_KEY, encryption_config)
+    kms_connection_config, crypto_factory = write_encrypted_file(
+        path, data_table, FOOTER_KEY_NAME, COL_KEY_NAME, FOOTER_KEY, COL_KEY,
+        encryption_config)
 
     verify_file_encrypted(path)
 
-    # Setup for reading
-    kms_connection_config, crypto_factory = setup_encryption_environment({
-        FOOTER_KEY_NAME: FOOTER_KEY.decode("UTF-8"),
-        COL_KEY_NAME: COL_KEY.decode("UTF-8"),
-    })
-
     # Read with decryption properties
     decryption_config = pe.DecryptionConfiguration(
         cache_lifetime=timedelta(minutes=5.0))
@@ -504,17 +501,12 @@ def test_encrypted_parquet_loop(tempdir, data_table, basic_encryption_config):
     # Encrypt the footer with the footer key,
     # encrypt column `a` and column `b` with another key,
     # keep `c` plaintext, defined in basic_encryption_config
-    write_encrypted_file(path, data_table, FOOTER_KEY_NAME, COL_KEY_NAME,
-                         FOOTER_KEY, COL_KEY, basic_encryption_config)
+    kms_connection_config, crypto_factory = write_encrypted_file(
+        path, data_table, FOOTER_KEY_NAME, COL_KEY_NAME, FOOTER_KEY, COL_KEY,
+        basic_encryption_config)
 
     verify_file_encrypted(path)
 
-    # Setup for reading with decryption properties
-    kms_connection_config, crypto_factory = setup_encryption_environment({
-        FOOTER_KEY_NAME: FOOTER_KEY.decode("UTF-8"),
-        COL_KEY_NAME: COL_KEY.decode("UTF-8"),
-    })
-
     decryption_config = pe.DecryptionConfiguration(
         cache_lifetime=timedelta(minutes=5.0))
     file_decryption_properties = crypto_factory.file_decryption_properties(
@@ -531,14 +523,9 @@ def test_encrypted_parquet_read_table(tempdir, data_table, basic_encryption_conf
     path = tempdir / PARQUET_NAME
 
     # Write the encrypted parquet file using the utility function
-    write_encrypted_file(path, data_table, FOOTER_KEY_NAME, COL_KEY_NAME,
-                         FOOTER_KEY, COL_KEY, basic_encryption_config)
-
-    # Setup for reading with decryption properties
-    kms_connection_config, crypto_factory = setup_encryption_environment({
-        FOOTER_KEY_NAME: FOOTER_KEY.decode("UTF-8"),
-        COL_KEY_NAME: COL_KEY.decode("UTF-8"),
-    })
+    kms_connection_config, crypto_factory = write_encrypted_file(
+        path, data_table, FOOTER_KEY_NAME, COL_KEY_NAME, FOOTER_KEY, COL_KEY,
+        basic_encryption_config)
 
     decryption_config = pe.DecryptionConfiguration(
         cache_lifetime=timedelta(minutes=5.0))
```

Reply:
I think 1 through 4 are addressed now.
As for 5 `decryption_config` and `file_decryption_properties` currently don't interact in Python code base. Perhaps @tolleybot has more insight.

Reply:
https://github.com/apache/arrow/blob/318d22adda3b66bd4a10fddc7789c8a13e4aa540/cpp/src/arrow/dataset/file_parquet_encryption_test.cc#L220-L233
It seems `DecryptionConfiguration` is meant for scanning, while `FileDecryptionProperties` is to be used when reading single files with `ReadTable`. Both seem to be public.